asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py

```diff
@@ -78,32 +78,44 @@ def provenance(self) -> dict[str, str]:
     def cache(
         prepped_complexes: list[PreppedComplex],
         dir: Union[str, Path],
-        type=CacheType.DesignUnit,
+        # type=CacheType.DesignUnit,
     ) -> None:
         """
-        Cache a set of design units for use later.
+        Cache the list of PreppedComplex in its own folder. Each is saved as a JSON, oedu PDB and ligand SDF for vis.
         """
         dir = Path(dir)
         if not dir.exists():
             dir.mkdir(parents=True)
 
         for pc in prepped_complexes:
-            if type == CacheType.DesignUnit:
-                du_name = pc.target.target_name + ".oedu"
-                du_path = dir / du_name
-                if not du_path.exists():
-                    pc.target.to_oedu_file(du_path)
-            elif type == CacheType.JSON:
-                json_name = pc.target.target_name + ".json"
-                json_path = dir / json_name
-                if not json_path.exists():
-                    pc.to_json_file(json_path)
+            # create a folder for the complex data
+            complex_folder = dir.joinpath(pc.target.target_name)
+            complex_folder.mkdir(parents=True)
+            pc.to_json_file(complex_folder.joinpath(pc.target.target_name + ".json"))
+            pc.target.to_oedu_file(
+                complex_folder.joinpath(pc.target.target_name + ".oedu")
+            )
+            pc.target.to_pdb_file(
+                complex_folder.joinpath(pc.target.target_name + ".pdb")
+            )
+            pc.ligand.to_sdf(complex_folder.joinpath(pc.ligand.compound_name + ".sdf"))
+
+            # if type == CacheType.DesignUnit:
+            #     du_name = pc.target.target_name + ".oedu"
+            #     du_path = dir / du_name
+            #     if not du_path.exists():
+            #         pc.target.to_oedu_file(du_path)
+            # elif type == CacheType.JSON:
+            #     json_name = pc.target.target_name + ".json"
+            #     json_path = dir / json_name
+            #     if not json_path.exists():
+            #         pc.to_json_file(json_path)
```

Comment:
so just to be clear - now every time we prep something, instead of deciding how we want to cache it (and only being able to pick one at a time), now we will cache it as all the types? I like this a lot actually

Reply:
Correct, I think the JSON should be the default as all other models are easily made from this and it's complete and usable downstream. We found loading from oedu was missing some data like a target name that made automation of later steps difficult. The pdb and sdf are mainly just for easy vis and could be made optional though.

Reply:
You could specify cache type multiple times but I agree this is just simpler overall at the slight cost of some disk space.